iso15118/shared/utils.py

```diff
@@ -34,8 +34,9 @@ def load_requested_protocols(read_protocols: Optional[List[str]]) -> List[Protoc
             f"No supported protocols configured. Supported protocols are "
             f"{supported_protocols} and could be configured in evcc_config.json"
         )
-    supported_protocols = [Protocol[x] for x in valid_protocols]
-    return supported_protocols
+    return [
+        Protocol[name] for name in supported_protocols if name in Protocol.__members__
+    ]
```

Comment:
```suggestion
    return [
        Protocol[name] for name in valid_protocols if name in Protocol.__members__
    ]
```
This should be valid_protocols . As else the full list would be returned?

Reply:
that's correct. Fixed